src/BcPrebidVast.js

```diff
@@ -73,16 +77,34 @@ function dispatchPrebidDoneEvent() {
 	document.dispatchEvent(event);
 }
 
-// the function loads prebid.js and does header bidding if needed
+// This function will be invoked immediately when plugin script is loaded in the page or by brightcove player.
+// The function loads prebid.js and does header bidding if needed.
+// The first parameter contains bidder settings needed for prebid.
+// If second parameter is present and is 'true' we have to call prebid right now (header bidding).
+// That parameter is present and set to 'true' when $$PREBID_GLOBAL$$.plugin_prebid_options.biddersSpec has a value.
+// $$PREBID_GLOBAL$$.plugin_prebid_options.biddersSpec has to be set for header bidding BEFORE the plugin's script is loaded.
 function loadPrebidScript(options, fromHeader) {
-    var pbjsScr = document.createElement('script');
-    pbjsScr.onload = function() {
-    	$$PREBID_GLOBAL$$.bc_pbjs = pbjs;
-    	// do header bidding if needed
+	// internal function which will be called later
+	var doInternalPrebid = function() {
+    	// do header bidding if fromHeader is 'true' and bidder setting are present in options
     	if (fromHeader && options && options.biddersSpec) {
-    		BC_prebid_in_progress = true;
+			BC_prebid_in_progress = true;
+			// invoke prebid
     		doPrebid(options, function(bids) {
-    			var arrBids = (bids && bids[options.biddersSpec.code]) ? bids[options.biddersSpec.code].bids : [];
+				// this function returns creative with higher CPM
+				var selectWinnerByCPM = function(arrBids) {
+					var cpm = 0.0;
+					var creative = null;
+					for (var i = 0; i < arrBids.length; i++) {
+						if (arrBids[i].cpm > cpm) {
+							cpm = arrBids[i].cpm;
```

Comment:
you might consider adjusting spacing to 2 spaces instead of 4 for deeply nested code readability.

Reply:
My code editor uses tab to control indentation and the size of a tab is aet globally to 4. We find that using 4 spaces for indentation makes the code more readable in general. Changing indentation spacing in particular areas will be inconvenient.